src/test/java/com/cloudogu/scm/ci/cistatus/service/CIStatusMergerTest.java

```diff
@@ -0,0 +1,147 @@
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package com.cloudogu.scm.ci.cistatus.service;
+
+import com.cloudogu.scm.ci.cistatus.workflow.SourceRevisionResolver;
+import com.cloudogu.scm.review.pullrequest.service.PullRequest;
+import com.cloudogu.scm.review.pullrequest.service.PullRequestService;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import sonia.scm.repository.Repository;
+import sonia.scm.repository.RepositoryTestData;
+
+import static com.cloudogu.scm.ci.cistatus.Constants.CHANGESET_STORE_NAME;
+import static com.cloudogu.scm.ci.cistatus.Constants.PULL_REQUEST_STORE_NAME;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(MockitoExtension.class)
+class CIStatusMergerTest {
+
+  private static final Repository REPOSITORY = RepositoryTestData.createHeartOfGold();
+
+  @Mock
+  private CIStatusService ciStatusService;
+
+  @Mock
+  private SourceRevisionResolver sourceRevisionResolver;
+
+  @Mock
+  private PullRequestService pullRequestService;
+
+  @InjectMocks
+  private CIStatusMerger merger;
+
+  @Test
+  void shouldReturnOnlyPullRequestCIStatusesIfBothCIStatusAvailable() {
+    String changesetId = "1a2b3c4d";
+    PullRequest pullRequest = new PullRequest();
+    pullRequest.setId("1");
+    pullRequest.setSource("develop");
+
+    String statusName = "JENKINS";
+    CIStatus prCIStatus = new CIStatus(statusName, statusName, statusName, Status.SUCCESS, "");
+    CIStatusCollection prCIStatusCollection = new CIStatusCollection();
+    prCIStatusCollection.put(prCIStatus);
+
+    CIStatus changesetCIStatus = new CIStatus(statusName, statusName, statusName, Status.FAILURE, "");
+    CIStatusCollection changesetCIStatusCollection = new CIStatusCollection();
+    changesetCIStatusCollection.put(changesetCIStatus);
+
+    when(pullRequestService.get(REPOSITORY, pullRequest.getId())).thenReturn(pullRequest);
+    when(sourceRevisionResolver.resolve(REPOSITORY, pullRequest.getSource())).thenReturn(changesetId);
+    when(ciStatusService.get(CHANGESET_STORE_NAME, REPOSITORY, changesetId)).thenReturn(changesetCIStatusCollection);
+    when(ciStatusService.get(PULL_REQUEST_STORE_NAME, REPOSITORY, pullRequest.getId())).thenReturn(prCIStatusCollection);
```

Comment:
These redundant code blocks are screeming for some helper methods to set up test data and behaviours.